client/src/components/SidePanel/Parameters/DynamicSwitch.tsx

```diff
@@ -35,20 +33,11 @@ function DynamicSwitch({
   });
 
   const selectedValue = useMemo(() => {
-    if (optionType === OptionTypes.Custom) {
-      // TODO: custom logic, add to payload but not to conversation
-      return inputValue;
-    }
-
     return conversation?.[settingKey] ?? defaultValue;
-  }, [conversation, defaultValue, optionType, settingKey, inputValue]);
+  }, [conversation, defaultValue, settingKey]);
```

Comment:
We could probably make this cleaner by doing: 

```typescript
const selectedValue = useMemo(() => 
  conversation?.[settingKey] ?? defaultValue, 
  [conversation?.[settingKey], defaultValue]
);
```

I might be wrong, I didn't do this in the editor but `conversation?.[settingKey]` will also only cause a rerender when the specific value changes instead of the whole object.

Reply:
In this case, I don't think useMemo is actually counterproductive as the overhead memoization and of tracking changes in dependencies costs a lot more than nullish coalescing operations. So we should change this to:

```ts
const selectedValue = conversation?.[settingKey] ?? defaultValue
```